### PR TITLE
Deprecate `config.Config` and `config.Service`, use `service.Config*`

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/4605

Followup PRs:
* Move service/internal/configunmarshaler as private struct into service/ to avoid circular dependency (no public API change)
* Change service/internal/builders to private structs in service/ or don't pass config, instead only pass the components and pipeline from the config.

While doing this, I realize that the "service" section in our config is a bit useless, we can think of promoting everything under "service::" one level up.

### CODE REVIEW

File: service/serviceconfig/service.go [hunk 1]
```diff
@@ -114,32 +120,16 @@ type ServiceTelemetryMetrics struct {
 	Address string `mapstructure:"address"`
 }
 
-func (s ServiceTelemetryMetrics) validate() error {
+func (s ServiceTelemetryMetrics) Validate() error {
```
Comment (Aneurysm9) [inline service/serviceconfig/service.go:123]: Do these need to be exported?  I don't see them used anywhere else.

Comment (bogdandrutu) [reply service/serviceconfig/service.go:123]: Done.

**Comment by Aneurysm9** [comment]
I don't think my questions and concerns from #4605 have been addressed.  I still don't know what the plan is here.

**Review by codeboten — COMMENTED** [review]
After reviewing the code in the PR I think the only question i have is the same as @Aneurysm9 mentioned before around what the plan to decouple `configunmarshaler` is

**Comment by bogdandrutu** [comment]
Converting this to draft, to block myself on merging, since we should do the backwards compatible change.

File: config/moved_service.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 )
 
 // Service defines the configurable components of the service.
+// Deprecated: Use serviceconfig.Service
```
Comment (codeboten) [inline config/moved_service.go:24]: The deprecation message needs a version number

**Review by codeboten — COMMENTED** [review]
>The reason is that in the "config" package after this change, there is no concept of "Service" so that packages offers utils for what it provides. The rest of the ConfigUnmarshaler will leave somewhere in the service package.

Right, in that case any reason not to move it as part of this change? This way we at least avoid introducing a dependency between on `serviceconfig` from a `config` package.

**Comment by bogdandrutu** [comment]
> This way we at least avoid introducing a dependency between on serviceconfig from a config package.

You will still have that since we need to deprecate things, unless we are willing to copy the code.

File: config/moved_service.go (hunk 2 above)
Comment (bogdandrutu) [reply config/moved_service.go]: Done.

**Comment by bogdandrutu** [comment]
> Right, in that case any reason not to move it as part of this change? This way we at least avoid introducing a dependency between on serviceconfig from a config package.

After some time, I got to the state you asked for. We don't have any circular dependency with `config` since the unmarshaler is now internal in the service. There is still a followup as documented in the description, but that is easy and non breaking to solve.

File: config/moved_service.go [hunk 3]
```diff
@@ -110,28 +114,13 @@ type ServiceTelemetryMetrics struct {
...
+// Deprecated: [v0.52.0] Use service.ConfigServicePipeline
 type Pipeline struct {
 	Receivers  []ComponentID `mapstructure:"receivers"`
 	Processors []ComponentID `mapstructure:"processors"`
 	Exporters  []ComponentID `mapstructure:"exporters"`
 }
 
-// Pipelines is a map of names to Pipelines.
-type Pipelines map[ComponentID]*Pipeline
+// Deprecated: will be removed soon.
```
Comment (codeboten) [inline config/moved_service.go:125]: Missing version

**Review by codeboten — APPROVED** [review]
Just one comment, otherwise the change looks good.

File: config/moved_service.go (hunk 3 above)
Comment (bogdandrutu) [reply config/moved_service.go]: Done.

**Pull request merged**